File: eclipse-workspace/CS284/src/shape/Rectangle.java
package shape;
/**
 * The class Rectangle models rectangles.
 * @author Zach
 *
 */

public class Rectangle extends Shape {
	//initiating variables
	private double base;
	private double height;
	
	public static int num_of_Rect;
	
	Rectangle(double b, double h, String c){
		super(c);
		height = h;
		base = b;
		
		
		num_of_Rect ++; //calculates the number of rectangles created
	}
	
	//Getters
	public double getBase() {
		return base;
	}
	
	public double getHeight() {
		return height;
	}
	
	//Setters
	/**
	 * Sets the base of a rectangle.
	 * @param b Base of a rectangle.
	 */
	public void setBase(double b) {
		base = b;
	}
	/**
	 * Sets the height of a rectangle.
	 * @param h Height of a rectangle.
	 */
	public void setHeight(double h) {
		height = h;
	}

	
	//methods
	public double area() {
		return base * height;
	}
	
	public String toString() {
		return "Base: " + base + " Height: " + height + " Color: " + color;
	}
	
	public Pair<Double,Double> getBaseHeight() {
		return new Pair<Double, Double>(base, height);
	}
}

File: eclipse-workspace/CS115A/rectangle.py
'''
Created on Nov 21, 2017

@author: Zach
'''
from shape import shape

class Rectangle(shape):
    def __init__(self, x, y, l, w, name = 'Rectangle'):
        super().__init__(x, y, name)
        self.__l = l 
        self.__w = w 
    
    @property
    def l(self):
        return self.__l
    @property
    def w(self):
        return self.__w
    @l.setter
    def l(self, l):
        self.__l = l 
    @w.setter 
    def w(self, w):
        self.__w = w 
     
    @property
    def area(self):
        return self.__l * self.__w
    
    def __str__(self):
        return super().__str__() + ', Length = ' + str(self.__l) + ', Width = ' + str(self.__w) + ', Area = ' + str(self.area)

if __name__ == '__main__':
    rect = Rectangle(10, 10, 20, 30)
    print(rect)
    
    
    print(isinstance(rect, Rectangle))
    print(isinstance(rect, shape))
    print(isinstance(rect, object))